solr/core/src/java/org/apache/solr/cluster/placement/plugins/AffinityPlacementFactory.java

```diff
@@ -242,6 +260,37 @@ public PlacementPlan computePlacement(Cluster cluster, PlacementRequest request,
         }
       }
 
+      if (secondaryCollection != null) {
```

Comment:
Can you extract the if bloc to a separate method?

Reply:
Also, comment on that method that the approach is a rather simplistic one, since the set of nodes chosen for the placement request does not take into account the existing `secondaryCollection` distribution, possibly leading to the creation of replicas for that collection that could have been avoided.
I assume clusters where such `withCollection` tricks are done can restrict placement of both collections to a set of nodes to better control what's going on...